bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVaneerSettingsFactory.java

```diff
@@ -0,0 +1,293 @@
+   * @throws IOException if any.
+   */
+  public static BigtableDataSettings createBigtableDataSettings(@Nonnull final BigtableOptions options) throws IOException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+    checkState(options.getRetryOptions().enableRetries(), "Disabling retries is not currently supported.");
+
+    final BigtableDataSettings.Builder builder = BigtableDataSettings.newBuilder();
+
+    builder.setProjectId(options.getProjectId());
+    builder.setInstanceId(options.getInstanceId());
+    builder.setAppProfileId(options.getAppProfileId());
+
+    builder.setEndpoint(options.getDataHost() + ":" + options.getPort());
+
+    builder.setCredentialsProvider(buildCredentialProvider(options.getCredentialOptions()));
+
+    buildBulkMutationsSettings(builder, options);
+
+    buildCheckAndMutateRowSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadModifyWriteSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadRowsSettings(builder, options);
+
+    buildMutateRowSettings(builder, options);
+
+    buildSampleRowKeysSettings(builder, options);
+
+    try {
+      ManagedChannel channel = BigtableSession
+          .createChannelPool(options.getDataHost(), options, options.getChannelCount());
+      builder.setTransportChannelProvider(
+          FixedTransportChannelProvider.create(GrpcTransportChannel.create(channel)));
+    } catch (GeneralSecurityException ex) {
+      throw new IOException("Could not create channels.", ex);
+    }
+
+    return builder.build();
+  }
+
+  /**
+   * To create an instance of {@link BigtableTableAdminSettings} from {@link BigtableOptions}.
+   *
+   * @param options a {@link BigtableOptions} object.
+   * @return a {@link BigtableTableAdminSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableTableAdminSettings createTableAdminSettings(@Nonnull final BigtableOptions options)
+      throws IOException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+
+    final BigtableTableAdminSettings.Builder adminBuilder = BigtableTableAdminSettings.newBuilder();
+
+    adminBuilder.setProjectId(options.getProjectId());
+    adminBuilder.setInstanceId(options.getInstanceId());
+
+    adminBuilder.stubSettings().setEndpoint(options.getAdminHost() + ":" + options.getPort());
+
+    adminBuilder.stubSettings()
+        .setCredentialsProvider(buildCredentialProvider(options.getCredentialOptions()));
+
+    try {
+      ManagedChannel channel = BigtableSession
+          .createChannelPool(options.getAdminHost(), options, options.getChannelCount());
+      adminBuilder.stubSettings().setTransportChannelProvider(
+          FixedTransportChannelProvider.create(GrpcTransportChannel.create(channel)));
```

Comment:
same thing as above but why this code is repeating? can we move to some common place and call this?